### PR TITLE
[maps][Expo Go] Update to latest

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

react-native-maps@1.10.0 fixes an issue with building for iOS simulators on Apple Silicon machines with Google Maps enabled

# How

```
et uvm -m react-native-maps -c "v1.10.0"
```

# Test Plan

Let CI run

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by ide — APPROVED** [review]
Tangentially - do we need to update our versions endpoint so that people install the right version of the JS for maps?

File: packages/expo/bundledNativeModules.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
   "react-native-web": "~0.19.6",
   "react-native-gesture-handler": "~2.14.0",
   "react-native-get-random-values": "~1.8.0",
-  "react-native-maps": "1.8.0",
+  "react-native-maps": "0.0.0",
```
Comment (tsapeta) [inline packages/expo/bundledNativeModules.json:91]: This doesn't seem correct 🤔 

```suggestion
  "react-native-maps": "1.10.0",
```

**Review by tsapeta — APPROVED** [review]

File: packages/expo/bundledNativeModules.json (hunk 1 above)
Comment (brentvatne) [reply packages/expo/bundledNativeModules.json:91]: yeah i'm not sure why uvm changed to 0.0.0 🤔

Comment (tsapeta) [reply packages/expo/bundledNativeModules.json:91]: Looks like they now have `0.0.0` in package.json 😅 
https://github.com/react-native-maps/react-native-maps/blob/4292c870fc7c82f1899a3e4a38f34912967d33d1/package.json#L8

**Pull request closed without merging**